zdm/H0_func_and_tests.py

```diff
@@ -0,0 +1,254 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Jun 15 18:53:22 2021
+
+@author: Esanmouli Ghosh
+"""
+import argparse
+
+import numpy as np
+from zdm import zdm
+#import pcosmic
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import matplotlib.cm as cm
+import matplotlib
+from pkg_resources import resource_filename
+import os
+import sys
+
+import scipy as sp
+import time
+from matplotlib.ticker import NullFormatter
+from zdm import iteration as it
+
+from zdm import survey
+from zdm import cosmology as cos
+from zdm import pcosmic
+from zdm import beams
+from zdm import misc_functions
+import pickle
+
+
+####setting up the initial grid and plotting some stuff####
+
+setH0=67.74
+cos.set_cosmology(H0=setH0)
+       
+# get the grid of p(DM|z)
+zDMgrid, zvals,dmvals,H0=misc_functions.get_zdm_grid(H0=setH0,new=True,plot=False,method='analytic')
+pcosmic.plot_mean(zvals,'Plots/mean_DM.pdf')
+    
+Wbins=10
+Wscale=2
+Nbeams=[20,20,20]  #Full beam NOT Std  
+thresh=0
+method=2
+Wlogmean=1.70267
+Wlogsigma=0.899148
+
+sdir = os.path.join(resource_filename('zdm', 'data'), 'Surveys/')
+#load the lat50 survey data
+lat50=survey.survey()
+lat50.process_survey_file(sdir+'parkes_mb_class_I_and_II.dat')
+DMhalo=50
+lat50.init_DMEG(DMhalo)
+lat50.init_beam(nbins=Nbeams[0],method=2,plot=False,thresh=thresh) # tells the survey to use the beam file
+pwidths,pprobs=survey.make_widths(lat50,Wlogmean,Wlogsigma,Wbins,scale=Wscale)
+efficiencies=lat50.get_efficiency_from_wlist(dmvals,pwidths,pprobs)
+    
+weights=lat50.wplist
+#misc_functions.plot_efficiencies(lat50)
+

```

Comment:
The functionality from lines 62 to 132 seems to be a combination of what is already within "initialise_grids" in "misc_functions.py", plus plots. I would suggest moving this to one call to "initialise grids" and for you to make another routine called "make_lots_of_plots". This shortens the bulk of the code into a few function calls and makes it easier to work out what is going on. If there is something in "initialise_grids" that needs a modification, then we should update that!

Reply:
Yes, most of this is actually inbuilt into the initialise_grids function - I just didn't use that in the beginning and diden't change it later. I should actually go back and use that function now. In my local update to this code, I have indeed added another "plots" routine to make the code look more manageable. I apologize for the mess here.